[package] fmt/all: test package fails with header_only option

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **fmt/5.3.0**
  * Operating System+version: **Linux Ubuntu 18.04**
  * Compiler+version: **Clang 8**
  * Conan version: **1.20.3**
  * Python version: **Python 3.7.4**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=clang
compiler.version=8
compiler.libcxx=libstdc++11
build_type=Release
[options]
[build_requires]
[env]
CC=/usr/bin/clang-8
CXX=/usr/bin/clang++-8

```

### Steps to reproduce (Include if Applicable)

conan create . fmt/5.3.0@ -o fmt:header_only=True

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
conan-center-index/recipes/fmt/all/test_package/test_package.cpp:55:21: error: no member named 'color' in namespace 'fmt'
    fmt::print(fmt::color::aqua, "Bincrafters\n");
               ~~~~~^
1 error generated.
CMakeFiles/test_package.dir/build.make:65: recipe for target 'CMakeFiles/test_package.dir/test_package.cpp.o' failed
make[2]: *** [CMakeFiles/test_package.dir/test_package.cpp.o] Error 1
make[2]: Leaving directory 'conan-center-index/recipes/fmt/all/test_package/build/5995dfbd1705ec21797daec978c7ce76d289bcdc'
CMakeFiles/Makefile2:80: recipe for target 'CMakeFiles/test_package.dir/all' failed
make[1]: *** [CMakeFiles/test_package.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 75%] Linking CXX executable bin/test_ranges
/home/andrew/.local/bin/cmake -E cmake_link_script CMakeFiles/test_ranges.dir/link.txt --verbose=1
/usr/bin/clang++-8   -m64 -stdlib=libstdc++ -O3 -DNDEBUG  -rdynamic CMakeFiles/test_ranges.dir/test_ranges.cpp.o  -o bin/test_ranges 
make[2]: Leaving directory conan-center-index/recipes/fmt/all/test_package/build/5995dfbd1705ec21797daec978c7ce76d289bcdc'
[ 75%] Built target test_ranges
make[1]: Leaving directory 'conan-center-index/recipes/fmt/all/test_package/build/5995dfbd1705ec21797daec978c7ce76d289bcdc'
Makefile:86: recipe for target 'all' failed
```

</details>

--- a/recipes/fmt/all/test_package/test_package.cpp
+++ b/recipes/fmt/all/test_package/test_package.cpp
@@ -4,13 +4,11 @@
 #include <vector>
 #include <limits>
 
-#ifdef FMT_HEADER_ONLY
-    #define FMT_EXTENDED_COLORS
-#endif
 
 #include "fmt/format.h"
 #include "fmt/printf.h"
 #include "fmt/ostream.h"
+#include "fmt/color.h"
 
 
 void vreport(const char *format, fmt::format_args args) {
@@ -51,9 +49,7 @@ int main() {
 
     fmt::print(std::cout, "{} {}\n", "Magic number", 42);
 
-#ifdef FMT_EXTENDED_COLORS
-    fmt::print(fmt::color::aqua, "Bincrafters\n");
-#endif
+    fmt::print(fg(fmt::color::aqua), "Bincrafters\n");
 
     return EXIT_SUCCESS;
 }

